src/main/groovy/com/bmuschko/gradle/docker/tasks/image/Dockerfile.groovy

```diff
@@ -31,23 +31,37 @@ class Dockerfile extends DefaultTask {
 
     @TaskAction
     void create() {
+        Instruction fromInstruction = findFromInstruction()
         verifyValidInstructions()
 
         getDestFile().withWriter { out ->
+            out.println fromInstruction.build()
             getInstructions().each { instruction ->
-                out.println instruction.build()
+                if(instruction.keyword != 'FROM'){ //FROM instruction already executed
+                    out.println instruction.build()
+                }
+            }
+        }
+    }
+
+    private Instruction findFromInstruction(){
+        Instruction fromInstruction
+        for(Instruction instruction: instructions){
+            if(instruction.keyword == 'FROM'){
+                fromInstruction = instruction
             }
         }
+        if(fromInstruction == null){
+            throw new IllegalStateException('Please specify a FROM instruction for your Dockerfile')
+        } else {
+            return fromInstruction
+        }
```

Comment:
If we return from inside the for loop we can just thrown the `IllegalStateException` if we reach this point.